pkg/controller/checker/checker.go

```diff
@@ -145,10 +145,15 @@ func (c *Checker) isLatestSemver(ctx context.Context, imageURL, currentSHA strin
 		isLatest = true
 	}
 
-	// If using the same image version, but the SHA has been updated upstream,
-	// make not latest
-	if currentImage.Equal(latestImageV) && currentSHA != latestImage.SHA && latestImage.SHA != "" {
+	// If using the same image version,
+	// but the SHA has been updated upstream,
+	// mark not latest
+	if currentImage.Equal(latestImageV) &&
+		!latestImage.MatchesSHA(currentSHA) &&
+		latestImage.SHA != "" {
```

Comment:
I think this will introduce a bug, if latestImage.SHA == "", but a child image does contain a SHA that is != to currentSHA, then isLatest is incorrectly true